### PR TITLE
Add --verbose flag to MLT build command.

### LINKED ISSUE #77: If a build takes longer than prior builds then start dumping output
Set a threshold beyond the prior build time and then start dumping output so user can see what's up

### PR DESCRIPTION
**Pull request by karthikvadla**
fixes #77 #39 

Adds `--verbose` flag to `mlt build`

usage:
`mlt build --verbose` or `mlt build -v`
## Reviewer Checklist

 - [ ] Do you understand it?
 - [ ] Is it correct?
 - [ ] Is it safe?
 - [ ] Is it legally compliant?
 - [ ] Is it robust?
 - [ ] Is it simple?
 - [ ] Is it tested?
 - [ ] Is it documented?
 - [ ] Is the style consistent?

See [the reviewer guide](docs/reviews.md) for more detail on each check.

### CODE REVIEW

File: mlt/commands/build.py [hunk 1]
```diff
@@ -50,20 +50,34 @@ def _build(self):
...
+        try:
+            build_process = process_helpers.run_popen(build_cmd, shell=True)
+
+            if self.args['--verbose']:
+                while True:
+                    output = build_process.stdout.readline()
+                    if output == '' and build_process.poll() is not None:
+                        error = build_process.stderr.readline()
+                        if error:
+                            raise Exception(error)
```
Comment (dmsuehir) [inline mlt/commands/build.py]: The way that we handle build errors doesn't seem consistent here.  In the non-verbose case we are printing the error in red and doing the `sys.exit(1)`.  In the verbose case, we are raising and Exception and then printing and doing the `sys.exit(1)` in the `except`

File: tests/unit/commands/test_build.py [hunk 2]
```diff
@@ -54,7 +54,9 @@ def test_simple_build(progress_bar_mock, popen_mock, open_mock, init_mock):
     progress_bar_mock.duration_progress.side_effect = \
         lambda x, y, z: print('Building')
 
-    build = BuildCommand({'build': True, '--watch': False})
+    build = BuildCommand({'build': True,
+                          '--watch': False,
+                          '--verbose': False})
```
Comment (dmsuehir) [inline tests/unit/commands/test_build.py]: Can we have a test for when --verbose is True?

File: mlt/commands/build.py (hunk 1 above)
Comment (s1113950) [reply mlt/commands/build.py]: :agree:

File: mlt/commands/build.py [hunk 3]
```diff
@@ -50,20 +50,34 @@ def _build(self):
...
-        if build_process.poll() != 0:
-            # When we have an error, get the stdout and error output
-            # and display them both with the error output in red.
-            output, error_msg = build_process.communicate()
-            print(output.decode("utf-8"))
-            print(colored(error_msg.decode("utf-8"), 'red'))
+        build_cmd = "CONTAINER_NAME={} make build".format(container_name)
+
+        try:
+            build_process = process_helpers.run_popen(build_cmd, shell=True)
```
Comment (s1113950) [inline mlt/commands/build.py]: if verbose mode couldn't you just set `stdout` and `stderr` to `True` here? Logic might be able to stay the same

File: tests/unit/commands/test_build.py (hunk 2 above)
Comment (s1113950) [reply tests/unit/commands/test_build.py]: :agree:

File: mlt/commands/build.py (hunk 3 above)
Comment (karthikvadla) [reply mlt/commands/build.py]: `build_process.communicate()` will be on hold, until entire process is completed. When docker image is large to build, it just hangs. The way i did will read each line continuously and display.

Comment (karthikvadla) [reply mlt/commands/build.py]:
```            build_process = process_helpers.run_popen(build_cmd,
                                                      shell=True)

            if self.args['--verbose']:
                build_process = process_helpers.run_popen(build_cmd,
                                                          shell=True,
                                                          stdout=True,
                                                          stderr=True)
            progress_bar.duration_progress(
                'Building', last_build_duration,
                lambda: build_process.poll() is not None)
            if build_process.poll() != 0:
                # When we have an error, get the stdout and error output
                # and display them both with the error output in red.
                output, error_msg = build_process.communicate()
                print(output.decode("utf-8"))
                print(colored(error_msg.decode("utf-8"), 'red'))
                sys.exit(1)``` 
You meant this way?

Comment (s1113950) [reply mlt/commands/build.py]: Yeah I figured that way could work. 👍 if that's not the case though

File: tests/unit/commands/test_build.py (hunk 2 above)
Comment (karthikvadla) [reply tests/unit/commands/test_build.py]: done.

File: mlt/commands/build.py [hunk 4]
```diff
@@ -61,10 +61,11 @@ def _build(self):
                     if output == '' and build_process.poll() is not None:
                         error = build_process.stderr.readline()
                         if error:
-                            raise Exception(error)
+                            print(colored(error.strip(), 'red'))
+                            sys.exit(1)
                         break
                     if output:
-                        print(output.strip().decode("utf-8"))
+                        print(output.strip())
```
Comment (s1113950) [inline mlt/commands/build.py]: You might still need `.decode(...` for python3 output

File: tests/unit/test_main.py [hunk 5]
```diff
@@ -42,9 +42,9 @@ def test_run_command(command):
 
 @pytest.mark.parametrize('args',
                          [{'<name>': 'Capitalized-Name',
-                           '-i': False, "-l": False, '--retries': '5'},
-                          {'-i': True, "-l": False, '<name>': 'foo', '--retries': '5'},
-                          {'-i': True, "-l": False, '<name>': 'foo', '--retries': '8'}])
+                           '-i': False, '-l': False, '-v': False, '--retries': '5'},
```
Comment (s1113950) [inline tests/unit/test_main.py]: Can there be a main unit test with verbose output as well?

File: tests/unit/commands/test_logs.py [hunk 6]
```diff
@@ -80,7 +80,7 @@ def test_logs_get_logs(json_mock, open_mock, verify_init, sleep_mock,
     log_value = '-'.join(['app', run_id])
     check_for_pods_readiness_mock.return_value = True
     process_helpers.return_value.stdout.readline.side_effect = [log_value, '']
-    process_helpers.return_value.poll.return_value = 1
+    process_helpers.return_value.poll.return_value = 0
```
[the remark this answers is not in the document]
Comment (karthikvadla) [reply tests/unit/commands/test_logs.py]: Actually that supposed to be `0` . Today i got to know, `poll()` exit code `0` indicates success and `1` indicates `error`.
```The poll() method will return

the exit code if the process is completed.
None if the process is still running.
```
https://www.endpoint.com/blog/2015/01/28/getting-realtime-output-using-python

File: tests/unit/test_main.py [hunk 7]
```diff
@@ -44,7 +44,9 @@ def test_run_command(command):
                          [{'<name>': 'Capitalized-Name',
                            '-i': False, '-l': False, '-v': False, '--retries': '5'},
                           {'-i': True, '-l': False, '-v': False, '<name>': 'foo', '--retries': '5'},
-                          {'-i': True, '-l': False, '-v': False, '<name>': 'foo', '--retries': '8'}])
+                          {'-i': True, '-l': False, '-v': False, '<name>': 'foo', '--retries': '8'},
+                          {'-i': True, '-l': False, '-v': False, '<name>': 'foo', '--retries': '8'},
```
Comment (s1113950) [inline tests/unit/test_main.py]: line 47 and 48 here are duplicates

File: tests/unit/test_main.py [hunk 8]
```diff
@@ -42,9 +42,11 @@ def test_run_command(command):
 
 @pytest.mark.parametrize('args',
                          [{'<name>': 'Capitalized-Name',
-                           '-i': False, "-l": False, '--retries': '5'},
-                          {'-i': True, "-l": False, '<name>': 'foo', '--retries': '5'},
-                          {'-i': True, "-l": False, '<name>': 'foo', '--retries': '8'}])
+                           '-i': False, '-l': False, '-v': False, '--retries': '5'},
+                          {'-i': True, '-l': False, '-v': False, '<name>': 'foo', '--retries': '5'},
+                          {'-i': True, '-l': False, '-v': False, '<name>': 'foo', '--retries': '8'},
+                          {'-i': True, '-l': False, '-v': False, '<name>': 'foo', '--retries': '8'},
```
Comment (s1113950) [inline tests/unit/test_main.py]: line 47 and line 48 are duplicates

**Pull request merged**